workflow/controller/operator.go

```diff
@@ -2188,3 +2191,56 @@ func (woc *wfOperationCtx) runOnExitNode(parentName, templateRef, boundaryID str
 	}
 	return false, nil, nil
 }
+
+func (woc *wfOperationCtx) createPDBResource() {
+
+	if woc.wf.Spec.PodDisruptionBudget == nil {
+		return
+	}
+	if woc.wf.Status.PDBResourceName != "" {
+		return
+	}
+
+	PDBName := woc.wf.Name + "-pdb"
+
+	pdb, _ := woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Get(PDBName, metav1.GetOptions{})
+	if pdb != nil && pdb.Name == PDBName {
+		return
+	}
+
+	if woc.wf.Spec.PodDisruptionBudget.Selector == nil {
+		woc.wf.Spec.PodDisruptionBudget.Selector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{common.LabelKeyWorkflow: woc.wf.Name},
+		}
+	}
+
+	newPDB := policyv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: woc.wf.Name + "-pdb",
+		},
+		Spec: *woc.wf.Spec.PodDisruptionBudget,
+	}
+	newPDB.SetLabels(make(map[string]string))
+	newPDB.Labels[common.LabelKeyWorkflow] = woc.wf.Name
+	newPDB.SetOwnerReferences(append(newPDB.GetOwnerReferences(), *metav1.NewControllerRef(woc.wf, wfv1.SchemeGroupVersion.WithKind(workflow.WorkflowKind))))
+	created, err := woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Create(&newPDB)
+	if err != nil {
+		woc.log.Errorf("Unable to create PDB resource for workflow, %s error: %s", woc.wf.Name, err)
+		return
+	}
+	woc.log.Infof("PDB resource %s created for workflow %s", created.Name, woc.wf.Name)
+	woc.wf.Status.PDBResourceName = created.Name
+	woc.updated = true
+}
+
+func (woc *wfOperationCtx) deletePDBResource() {
+	if woc.wf.Status.PDBResourceName == "" {
+		return
+	}
+	err := woc.controller.kubeclientset.PolicyV1beta1().PodDisruptionBudgets(woc.wf.Namespace).Delete(woc.wf.Status.PDBResourceName, &metav1.DeleteOptions{})
```

Comment:
you could use `DeleteCollection` with `ListOptions(labelSelector='workflowName=...`